spring-boot-examples/workflows/src/test/java/io/dapr/springboot/examples/wfp/WorkflowPatternsAppTests.java

```diff
@@ -0,0 +1,142 @@
+        webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class WorkflowPatternsAppTests {
+
+  @Autowired
+  private DaprClient daprClient;
+
+  @BeforeEach
+  void setUp() {
+    RestAssured.baseURI = "http://localhost:" + 8080;
+    org.testcontainers.Testcontainers.exposeHostPorts(8080);
+  }
+
+
+  @Test
+  void testChainWorkflow() {
+    given().contentType(ContentType.JSON)
+            .body("")
+            .when()
+            .post("/wfp/chain")
+            .then()
+            .statusCode(200).body(containsString("TOKYO, LONDON, SEATTLE"));
+  }
+
+  @Test
+  void testChildWorkflow() {
+    given().contentType(ContentType.JSON)
+            .body("")
+            .when()
+            .post("/wfp/child")
+            .then()
+            .statusCode(200).body(containsString("!wolfkroW rpaD olleH"));
+  }
+
+  @Test
+  void testFanOutIn() {
+    List<String> listOfStrings = Arrays.asList(
+            "Hello, world!",
+            "The quick brown fox jumps over the lazy dog.",
+            "If a tree falls in the forest and there is no one there to hear it, does it make a sound?",
+            "The greatest glory in living lies not in never falling, but in rising every time we fall.",
+            "Always remember that you are absolutely unique. Just like everyone else.");
+
+    given().contentType(ContentType.JSON)
+            .body(listOfStrings)
+            .when()
+            .post("/wfp/fanoutin")
+            .then()
+            .statusCode(200).body("wordCount",equalTo(60));
+  }
+
+  @Test
+  void testExternalEventApprove() {
+
+    given()
+            .queryParam("orderId", "123")
+            .when()
+            .post("/wfp/externalevent")
+            .then()
+            .statusCode(200).extract().asString();
+
+
+
+    given()
+            .queryParam("orderId", "123")
+            .queryParam("decision", true)
+            .when()
+            .post("/wfp/externalevent-continue")
+            .then()
+            .statusCode(200).body("approved", equalTo(true));
+  }
+
+  @Test
+  void testExternalEventDeny() {
+
+    given()
+            .queryParam("orderId", "123")
+            .when()
+            .post("/wfp/externalevent")
+            .then()
+            .statusCode(200).extract().asString();
+
+
+
+    given()
+            .queryParam("orderId", "123")
+            .queryParam("decision", false)
+            .when()
+            .post("/wfp/externalevent-continue")
+            .then()
+            .statusCode(200).body("approved", equalTo(false));
+  }
+
+
+  @Test
+  void testContinueAsNew() {
+    //This call blocks until all the clean up activities are executed
+    CleanUpLog cleanUpLog = given().contentType(ContentType.JSON)
+            .body("")
+            .when()
+            .post("/wfp/continueasnew")
+            .then()
+            .statusCode(200).extract().as(CleanUpLog.class);
+
+    assertEquals(5, cleanUpLog.getCleanUpTimes());
+  }
+
```

Comment:
can you delete this if its an empty file: `spring-boot-examples/workflows/src/test/resources/application.properties`?

Reply:
This is something that we do in spring applications, if you want to customize the app configuration, that file needs to be there to accept properties. Instead of deleting it, I will add the spring boot app name to follow what https://start.spring.io/ is doing.

Reply:
I believe the file was empty when I commented about it initially... I see a value in it now, so all good 👍🏻